ws-service/server.ts

```diff
@@ -70,11 +72,127 @@ const broadcastRouteConnection = (socket: Socket): void => {
     socket.disconnect(true)
     return
   }
-  void socket.on('txs-broadcast', broadcastTxs)
+  void socket.on(SOCKET_MESSAGES.TXS_BROADCAST, broadcastTxs)
+}
+
+const sendSideshiftPairRate = async (getPairRateData: GetPairRateData): Promise<SideshiftPairRes> => {
+  const res = await fetch(BASE_SIDESHIFT_URL + `pair/${getPairRateData.from}/${getPairRateData.to}`)
+  const data = await res.json()
+  return data as SideshiftPairRes
+}
+
+type ErrorType = 'quote-error' | 'shift-error'
+interface SideshiftError {
+  errorType: ErrorType
+  errorMessage: string
+}
+
+const createQuote = async (createQuoteData: CreateQuoteAndShiftData): Promise<SideshiftQuoteRes | SideshiftError> => {
+  const requestBody = JSON.stringify({
+    ...createQuoteData,
+    affiliateId: config.sideshiftAffiliateId
+  })
+
+  const res = await fetch(BASE_SIDESHIFT_URL + 'quotes', {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json',
+      'x-sideshift-secret': process.env.SIDESHIFT_SECRET_KEY as string
+    },
+    body: requestBody
+  })
+  const data = await res.json()
+  if ('error' in data) {
+    console.error('Error when creating sideshift quote.', {
+      createQuoteData,
+      responseData: data
+    })
+    return {
+      errorType: 'quote-error',
```

Comment:
The thing is, since this is being defined already as a template literal type, in the line
```
type SideshiftErrorType = 'quote-error' | 'shift-error'
```
... this string already works as a constant, so e.g. the LSP is able to see what is possible as demonstrated in the screenshot below:
![image](https://github.com/user-attachments/assets/3ecbbbde-d5d7-4f10-8e70-cf643e7e1041)

it's a standard way of doing things in typescript, Chronik TS module uses that e.g. in the `WsMsgClient` `msg` and `msgType` attribute, you can see it in `services/chronikService.ts:349`